crawlerr/services/strategies/directory_strategy.py

```
@@ -0,0 +1,500 @@
+                markdown=content if file_path.suffix.lower() == ".md" else None,
+                links=[],  # Could extract file references in future
+                images=[],
+                metadata=metadata,
+                timestamp=time.time(),
+                word_count=metadata["word_count"],
+            )
+
+            return page_content
+
+        except Exception as e:
+            raise Exception(f"Failed to process file {file_path}: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)